src/rgw/rgw_sal.h

```diff
@@ -1170,6 +1170,10 @@ class MultipartPart {
  */
 class MultipartUpload {
 public:
+  //object lock
+  std::optional<RGWObjectRetention> obj_retention = std::nullopt;
+  std::optional<RGWObjectLegalHold> obj_legal_hold = std::nullopt;
```

Comment:
i don't think these belong as member variables in the base class

can they be passed as arguments to `init()` and returned as output params by `get_info()` instead?